src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1689,6 +1710,27 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
                 }
             }
 
+            if (win32Resources != null)
+            {
+                if (!win32Resources.CanRead || !win32Resources.CanSeek)
+                {
+                    throw new ArgumentException(CodeAnalysisResources.StreamMustSupportReadAndSeek, nameof(win32Resources));
+                }
+            }
+
+            if (sourceLinkStream != null)
+            {
+                if (options?.DebugInformationFormat == DebugInformationFormat.Pdb)
+                {
+                    throw new ArgumentException(CodeAnalysisResources.SourceLinkRequiresPortablePdb, nameof(sourceLinkStream));
+                }
+
+                if (!sourceLinkStream.CanRead)
+                {
+                    throw new ArgumentException(CodeAnalysisResources.StreamMustSupportRead, nameof(sourceLinkStream));
```

Comment:
[https://github.com/dotnet/roslyn/pull/12854/files/0b37c814dfca154377c8a7c3b56b68012f54570c#diff-e3a813804251fde5bef8a02606a462cdR288](https://github.com/dotnet/roslyn/pull/12854/files/0b37c814dfca154377c8a7c3b56b68012f54570c#diff-e3a813804251fde5bef8a02606a462cdR288)

---
In reply to: [73059322](https://github.com/dotnet/roslyn/pull/12854#discussion_r73059322) [](ancestors = 73059322)